### ISSUE TITLE
File selection window doesn't get pushed to back

### ISSUE THREAD

**Issue by marijnvdwerf** [OpenRCT2/OpenRCT2#2763]
Load file window always gets rendered on top. Other windows get pushed to the back when a new window is selected. This can be observed in the start screen.

1. Press 'load' button
2. Press scenario button

Expected:
Scenario window should be fully visible

Actual:
Scenario window stays behind load window.

**Comment by Gymnasiast** [comment]
These actions are mutually exclusive, so opening one should close the other, AFAIC.

**Comment by IntelOrca** [comment]
Load / Save window should be a modal window which stops all other windows from working until its closed.

**Comment by marijnvdwerf** [comment]
If you're going for that direction, _please_ make it an option to disable the in-game window :/

**Comment by Gymnasiast** [comment]
Why?

**Comment by IntelOrca** [comment]
So who ever feels like coding this... I suggest a couple of new variables in input.c called gModalWindowClass and gModalWindowNumber to identify which window is set to be modal. Then have all input actions only work for the modal window if set. A class of 255 can mean no modal window.

**Comment by marijnvdwerf** [comment]
@Gymnasiast Because it's a fairly inconvenient way to browse the file system/open stuff. It _works_, but it's fairly clumsy. Like picking a folder using this thing:

![](http://www.yevol.com/en/dlgboxes/browse1.gif)

Using the native file browser you can easily go to your 'pinned' folders, and even drag a file on it to immediately select it.

**Comment by Gymnasiast** [comment]
I'm not really sold on the idea of model windows. Simply closing all other windows when opening the load/save or scenario window should do the trick IMO.

@marijnvdwerf : There is a button to open the native system dialog if you prefer that.

**Comment by marijnvdwerf** [comment]
@Gymnasiast I know. The only benefit of the current window is that it's an in-game window. Turning it into a modal will make it nothing better than a native modal.

**Comment by pizza2004** [comment]
I think the in game window could stand to be improved, but I don't know if I like the modal window idea either.

**Comment by IntelOrca** [comment]
@Gymnasiast so when you want to save the game, it should close all your windows?

**Comment by marijnvdwerf** [comment]
Should the in-game behavior match the title screen behavior? There's no reason for the title screen to actually have windows, besides code-reuse.

**Comment by Gymnasiast** [comment]
@IntelOrca : Mutually exclusive windows, like the scenario selector at least. And the window comes on top anyway, burying it under other windows requires action.

AFAIC, the scenario selector should close the load/save window, and vice versa. I think that should be enough.

**Comment by IntelOrca** [comment]
But in-game, you don't want it to close all your windows.

**Comment by Gymnasiast** [comment]
No, but that isn't necessary. (I know I suggested it earlier, but I retract that.)
Just having the scenario selector close the load/save window, and vice versa should avoid confusion.

**Comment by IntelOrca** [comment]
OK, but what about the modal idea for in-game?
Otherwise you can have the load save window open while constructing a ride or something.

**Comment by Gymnasiast** [comment]
Is there a case when it's bad to have those two open at the same time?

**Comment by IntelOrca** [comment]
Most applications disable the application when showing a load or save window until that window is closed.

**Comment by marijnvdwerf** [comment]
Most applications don't have full window management inside themselves ;)

Anyhow, having it act as a modal __while in the game__ would be a benefit,
because otherwise you'd lose the context of what the save/load window is
for.
Op ma 18 jan. 2016 om 19:50 schreef Ted John <notifications@github.com>

**Comment by Broxzier** [comment]
A modal approach could also be used for confirmation boxes when you exit a game without having saved recently.

Though I'm also with Gymnasiast on this one, and let it close only certain windows that shouldn't be open at the same time.

**Comment by pizza2004** [comment]
I actually agree with IntelOrca now.  I'd much rather have it be modal than have people dragging it around while in the game.

**Comment by jensj12** [comment]
It's even irritating me to see the game running in the background while I have the save window open (everytime I still expect it to pause, even though it just doesn't). Please make it modal.

**Comment by IntelOrca** [comment]
The red transparent close / without saving? window should also be modal and pause the game.

Unless its multiplayer in which case, don't pause the game.

**Comment by Broxzier** [comment]
In that case perhaps mute the sounds? If we're going for modal, I'd like to have the escape key close the window.

**Comment by marijnvdwerf** [comment]
Maybe instead of muting you could add a muffled effect over the sound
